roles/openshift_master/tasks/system_container.yml

```diff
@@ -26,11 +27,51 @@
   when:
     - ("master" in result.stdout)
     - (openshift.common.version is not defined) or (openshift.common.version != openshift_version) | bool
+    - openshift.master.ha is not defined or not openshift.master.ha | bool
 
 - name: Install Master system container package
   command: >
     atomic install --system --name={{ openshift.common.service_type }}-master {{ openshift.common.system_images_registry }}/{{ openshift.master.master_system_image }}:{{ openshift_image_tag }}
   when:
     - (openshift.common.version is not defined) or (openshift.common.version != openshift_version) or ("master" not in result.stdout) | bool
+    - openshift.master.ha is not defined or not openshift.master.ha | bool
+  notify:
+    - restart master
```

Comment:
the problem is that each of them have a different condition, and it seems I cannot `when` inside of a block.  I can try that again.

Anyway, I consider this PR in addition to the previous one a temporary solution just to get it feature complete as soon as possible so that we can figure out if there is still "something big" missing.

At that point, I'd like to rework this file completely (e.g. idempotent updates also between different versions, so that we can drop the uninstall/install logic).

I'd like to not check at all the version here, I want to be able to simply do "atomic update containers $CONTAINER --to=$VERSION" and let atomic understand when there is nothing to do.